packages/utils/test/crypto.spec.ts

```diff
@@ -108,4 +112,28 @@ describe("Crypto", () => {
   it("calls generateRandomBytes32", () => {
     expect(generateRandomBytes32()).toBeTruthy();
   });
+  it("should validate verify v2 jwt", async () => {
```

Comment:
Would love to see more test cases here, specially for unhappy paths and also for the timeout logic.